docker/Dockerfile

```diff
@@ -0,0 +1,130 @@
+USER root
+RUN if [ "HOST_GID" != "1000"] ; \
+    then groupadd --gid ${HOST_GID} host_group && \
+    usermod ${USER} -g ${HOST_GID} && \ 
+    usermod ${USER} -a -G ${USER_GROUP}; fi
+USER ${USER}
+
+# Setup git identity
+RUN git config --global user.name "${GIT_NAME}"
+RUN git config --global user.email "${GIT_EMAIL}"
+
+# Setup python version for noetic
+RUN sudo apt update
+RUN if [ "${ROS_DISTRO}" == "noetic" ] ; \
+    then sudo apt install python-is-python3 ; fi
+
+# Handle SIMD option
+RUN if [ "${USE_SIMD}" = "ON" ] ; \ 
+    then export CMAKE_CXX_FLAGS="-march=native -faligned-new" ; fi
+
+### Install all dependencies of IIWA ROS
+# Clone KUKA FRI (need to be root to clone private repo)
+WORKDIR /tmp
+USER root
+RUN --mount=type=ssh git clone git@github.com:epfl-lasa/kuka_fri.git
+WORKDIR /tmp/kuka_fri
+RUN if [ "${USE_SMID}" != "ON" ] ; \
+    then wget https://gist.githubusercontent.com/matthias-mayr/0f947982474c1865aab825bd084e7a92/raw/244f1193bd30051ae625c8f29ed241855a59ee38/0001-Config-Disables-SIMD-march-native-by-default.patch \
+    ; fi
+RUN --mount=type=ssh  if [ "${USE_SMID}" != "ON" ] ; \
+    then git am 0001-Config-Disables-SIMD-march-native-by-default.patch ; fi
+
+# Transfer repo back to original user after root clone
+WORKDIR /tmp
+RUN chown -R ${USER}:${HOST_GID} kuka_fri
+
+# Install kuka Fri as USER
+USER ${USER}
+RUN cd kuka_fri && ./waf configure && ./waf && sudo ./waf install
+
+# Install SpaceVecAlg
+RUN git clone --recursive https://github.com/jrl-umi3218/SpaceVecAlg.git
+RUN cd SpaceVecAlg && mkdir build && cd build \
+    && cmake -DCMAKE_BUILD_TYPE=Release -DPYTHON_BINDING=OFF .. \
+    && make -j && sudo make install
+
+# Install RBDyn
+RUN git clone --recursive https://github.com/jrl-umi3218/RBDyn.git
+RUN cd RBDyn && mkdir build && cd build \
+    && cmake -DCMAKE_BUILD_TYPE=Release -DPYTHON_BINDING=OFF .. \
+    && make -j && sudo make install
+
+# Install mc_rbdyn_urdf
+RUN git clone --recursive -b v1.1.0 https://github.com/jrl-umi3218/mc_rbdyn_urdf.git
+RUN cd mc_rbdyn_urdf && mkdir build && cd build \
+    && cmake -DCMAKE_BUILD_TYPE=Release -DPYTHON_BINDING=OFF .. \
+    && make -j && sudo make install
+
+# Install corrade
+RUN git clone https://github.com/mosra/corrade.git
+RUN cd corrade && git checkout 0d149ee9f26a6e35c30b1b44f281b272397842f5 \
+    && mkdir build && cd build && cmake .. && make -j && sudo make install
+
+# Install robot_controller
+RUN git clone https://github.com/epfl-lasa/robot_controllers.git
+RUN cd robot_controllers && mkdir build && cd build \
+    && cmake .. && make -j && sudo make install
+
+# Remove temporari files
+RUN sudo ldconfig
+RUN rm -rf /tmp/*
+
+### Install gazebo
+# Install gazebo (9 or 11 depending on distro)
+WORKDIR /home/${USER}
+RUN sudo apt-get update
+RUN if [ "$ROS_DISTRO" = "noetic" ] ; then sudo apt-get install -y gazebo11 ; fi
+RUN if [ "$ROS_DISTRO" = "melodic" ] ; then sudo apt-get install -y gazebo9 ; fi
+
+# Install gaezbo ros packages
```

Comment:
I would usually install all apt packages before cloning. If you ever need to do the cloning again because the branch has new features, it will install gazebo again (and take longer)  than if you put it before to optimize caching

Reply:
Cool tip, thanks